lib/datadog/search-syntax/src/vrl.rs

```diff
@@ -0,0 +1,549 @@
+                (ComparisonValue::Unbounded, ComparisonValue::Unbounded) => {
+                    make_function_call("exists", vec![make_query(attr)])
+                }
+                // Unbounded lower. Wrapped in a container group for negation compatibility.
+                (ComparisonValue::Unbounded, _) => make_container_group(make_op(
+                    make_node(make_query(attr)),
+                    if upper_inclusive {
+                        ast::Opcode::Le
+                    } else {
+                        ast::Opcode::Lt
+                    },
+                    make_node(ast::Expr::Literal(make_node(upper.into()))),
+                )),
+                // Unbounded upper. Wrapped in a container group for negation compatibility.
+                (_, ComparisonValue::Unbounded) => make_container_group(make_op(
+                    make_node(make_query(attr)),
+                    if lower_inclusive {
+                        ast::Opcode::Ge
+                    } else {
+                        ast::Opcode::Gt
+                    },
+                    make_node(ast::Expr::Literal(make_node(lower.into()))),
+                )),
+                // Definitive range.
+                _ => make_container_group(make_op(
+                    make_node(make_op(
+                        make_node(make_query(attr.clone())),
+                        if lower_inclusive {
+                            ast::Opcode::Ge
+                        } else {
+                            ast::Opcode::Gt
+                        },
+                        make_node(ast::Expr::Literal(make_node(lower.into()))),
+                    )),
+                    ast::Opcode::And,
+                    make_node(make_op(
+                        make_node(make_query(attr)),
+                        if upper_inclusive {
+                            ast::Opcode::Le
+                        } else {
+                            ast::Opcode::Lt
+                        },
+                        make_node(ast::Expr::Literal(make_node(upper.into()))),
+                    )),
+                )),
+            },
+            // Negation.
+            QueryNode::NegatedNode { node } => make_not(ast::Expr::from(*node)),
+            // Compound.
+            QueryNode::Boolean { oper, nodes } => {
+                make_container_group(nest_exprs(nodes.into_iter(), oper))
+            }
+        }
+    }
+}
+
+/// Creates a VRL node with a default span.
+fn make_node<T>(node: T) -> ast::Node<T> {
+    ast::Node::new(Span::default(), node)
+}
+
+/// Converts a field/facet name to the VRL equivalent. Datadog payloads have a `message` field
+/// (which is used whenever the default field is encountered. Facets are hosted on .custom.*.
+fn normalize_tag<T: AsRef<str>>(value: T) -> String {
+    let value = value.as_ref();
+    if value.eq(grammar::DEFAULT_FIELD) {
+        return "message".to_string();
```

Comment:
Actually it's not only message but also:
"custom.error.message",
"custom.error.stack",
"custom.title"

I guess it complicates things a bit, since we need to check all of them.

Reply:
😄 yeah, that's definitely good to know. So to confirm -- when there's _no_ tag/facet specified, it should search _all_ of the above via an `OR` match -- is that right?

To confirm how that'd look in VRL -- a search for:

```
bla
```

Would be translated to the equivalent of:

```
match(.message, r'\bbla\b') || match(.custom.error.message, r'\bbla\b') || match(.custom.error.stack, r'\bbla\b') || match(.custom.title, r'\bbla\b') || match(._default_, r'\bbla\b')
```

(Or some other function that could search on multiple keys)